segment/reader.go

```diff
@@ -114,9 +114,9 @@ func (r *Reader) makeBuffer() *types.PooledBuffer {
 	if r.bufPool == nil {
 		return &types.PooledBuffer{Bs: make([]byte, minBufSize)}
 	}
-	buf := r.bufPool.Get().([]byte)
+	buf := r.bufPool.Get().(*[]byte)
 	return &types.PooledBuffer{
-		Bs: buf,
+		Bs: *buf,
```

Comment:
With this change the following error is fixed:

```
raft-wal/segment/reader.go:123:18: SA6002: argument should be pointer-like to avoid allocations (staticcheck)
                        r.bufPool.Put(buf)
```

Verified that `bufPool.Get()` and `Put()` is not called anywhere else.